pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -90,9 +90,35 @@ type ArtifactGCStrategy string
 const (
 	ArtifactGCOnWorkflowCompletion ArtifactGCStrategy = "OnWorkflowCompletion"
 	ArtifactGCOnWorkflowDeletion   ArtifactGCStrategy = "OnWorkflowDeletion"
+	ArtifactGCOnWorkflowSuccess    ArtifactGCStrategy = "OnWorkflowSuccess"
+	ArtifactGCOnWorkflowFailure    ArtifactGCStrategy = "OnWorkflowFailure"
 	ArtifactGCNever                ArtifactGCStrategy = ""
 )
 
+var AnyArtifactGCStrategy = map[ArtifactGCStrategy]bool{
+	ArtifactGCOnWorkflowCompletion: true,
+	ArtifactGCOnWorkflowDeletion:   true,
+	ArtifactGCOnWorkflowSuccess:    true,
+	ArtifactGCOnWorkflowFailure:    true,
+}
+
+func (s ArtifactGCStrategy) AbbreviatedName() string {
+	switch s {
+	case ArtifactGCOnWorkflowCompletion:
+		return "wfcomp"
```

Comment:
what is this used for? is it really needed? could you just use the name? might be clearer

Reply:
it's being used in the pod name - I thought that the full name would make the pod name rather long but can use it if you prefer. I also decided against hashing this since it's nice to know by looking at the pod name what it is.